forward_command_controller/include/forward_command_controller/forward_joint_group_command_controller.h

```diff
@@ -106,33 +99,87 @@ class ForwardJointGroupCommandController: public controller_interface::Controlle
 
     commands_buffer_.writeFromNonRT(std::vector<double>(n_joints_, 0.0));
 
-    sub_command_ = n.subscribe<std_msgs::Float64MultiArray>("command", 1, &ForwardJointGroupCommandController::commandCB, this);
     return true;
   }
 
-  void starting(const ros::Time& time);
-  void update(const ros::Time& /*time*/, const ros::Duration& /*period*/)
+  virtual void starting(const ros::Time& time);
+  virtual void update(const ros::Time& /*time*/, const ros::Duration& /*period*/)
   {
     std::vector<double> & commands = *commands_buffer_.readFromRT();
     for(unsigned int i=0; i<n_joints_; i++)
     {  joints_[i].setCommand(commands[i]);  }
   }
 
+protected:
   std::vector< std::string > joint_names_;
   std::vector< hardware_interface::JointHandle > joints_;
   realtime_tools::RealtimeBuffer<std::vector<double> > commands_buffer_;
   unsigned int n_joints_;
+};
+
+template <> inline void ForwardJointGroupCommandControllerBase<hardware_interface::EffortJointInterface>::starting(const ros::Time& time)
```

Comment:
I think moving these three inline functions back into their cpp files makes sense:
```
effort_controllers/src/joint_group_effort_controller.cpp
position_controllers/src/joint_group_position_controller.cpp
velocity_controllers/src/joint_group_velocity_controller.cpp
```
The benefit here is that having to include their respective `.h` files protects us against prematurely specialization these functions. For instance, I think we need 
```
effort_controllers::JointGroupPositionController
effort_controllers::JointGroupVelocityController
```
which would be the `group` analogs of
[effort_controllers::JointPositionController](https://github.com/ros-controls/ros_controllers/blob/indigo-devel/effort_controllers/include/effort_controllers/joint_position_controller.h)
[effort_controllers::JointVelocityController](https://github.com/ros-controls/ros_controllers/blob/indigo-devel/effort_controllers/include/effort_controllers/joint_velocity_controller.h)

both of which would inherit from `ForwardJointGroupCommandControllerBase<hardware_interface::EffortJointInterface>` but will likely require different `starting(const ros::Time& time)` implementations from `effort_controllers::JointGroupEffortController::starting()` (ie. the above line)